exotations/exotica_core_task_maps/src/look_at.cpp

```diff
@@ -70,47 +55,22 @@ void LookAt::Update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi, Eigen::Ma
 
     for (int i = 0; i < n_end_effs_; ++i)
     {
-        const int end_effector_id = i * n_end_effs_;
-
-        // Get EffPoint and LookAtTarget
-        Eigen::Vector3d c = Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(end_effector_id).p.data);      // EffPoint | Eff frame
-        Eigen::Vector3d p = Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(end_effector_id + 1).p.data);  // LookAtTarget | Eff frame
-
-        // Compute orthogonal orthogonal projection a onto line e->c
-        double c_squared_norm = c.squaredNorm();
-        double alpha = p.dot(c) / c_squared_norm;
-        Eigen::Vector3d a = alpha * c;
-
-        // Set Phi
-        phi.segment<3>(end_effector_id) = a - p;
-
-        // Compute jacobian
+        Eigen::Vector3d p = Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(i).p.data);
+        p(2) = 0.0;
+        phi.segment<3>(i * 3) = p;
         for (int j = 0; j < jacobian.cols(); ++j)
         {
-            Eigen::Vector3d pd = kinematics[0].jacobian[end_effector_id + 1].data.topRows<3>().col(j);
-            double alphad = c.dot(pd) / c_squared_norm;
-            jacobian.middleRows<3>(end_effector_id).col(j) = alphad * c - pd;
+            Eigen::Vector3d pd = kinematics[0].jacobian[i].data.topRows<3>().col(j);
+            pd(2) = 0.0;
+            jacobian.middleRows<3>(i).col(j) = pd;
         }
     }
 }
 
 void LookAt::Instantiate(const LookAtInitializer& init)
 {
-    // Error check
-    if (frames_.size() % 3 != 0) ThrowNamed("Three frames are required for each end-effector!");
-
-    // Init private variables
-    n_end_effs_ = frames_.size() / 3;
-    n_ = frames_.size();
-
-    // Verify that the second and third frames are the same
-    for (int i = 0; i < n_end_effs_; ++i)
-    {
-        if (frames_[i + 1].frame_A_link_name != frames_[i + 2].frame_A_link_name)
-        {
-            ThrowPretty("The second and third links (LookAtTarget) need to be the same! Got: " << frames_[i + 1].frame_A_link_name << " and " << frames_[i + 2].frame_A_link_name);
-        }
-    }
+    n_end_effs_ = frames_.size();
+    n_ = n_end_effs_ * 3;
```

Comment:
Technically it could be times two - and above instead of setting it to zero for the z-dimension, just omit it. Same for Jacobian - you can just get top-2 instead of top-3.